types/typeBuilders.d.ts

```diff
@@ -0,0 +1,66 @@
+import {NodeTypes, PortTypes} from "./index";
+
+export namespace Controls {
+    function text(config: any): {
+        type: string;
+        label?: string;
+        name: string;
+        defaultValue: any;
+        setValue?: (oldData: any, newData: any) => any;
+    };
+    function select(config: any): {
+        type: string;
+        label?: string;
+        name: string;
+        defaultValue: any;
+        setValue?: (oldData: any, newData: any) => any;
+    };
+    function number(config: any): {
+        type: string;
+        label?: string;
+        name: string;
+        defaultValue: any;
+        setValue?: (oldData: any, newData: any) => any;
+    };
+    function checkbox(config: any): {
+        type: string;
+        label?: string;
+        name: string;
+        defaultValue: any;
+        setValue?: (oldData: any, newData: any) => any;
+    };
+    function multiselect(config: any): {
+        type: string;
+        label?: string;
+        name: string;
+        defaultValue: any;
+        setValue?: (oldData: any, newData: any) => any;
+    };
+    function custom(config: any): {
+        type: string;
+        label?: string;
+        name: string;
+        defaultValue: any;
+        setValue?: (oldData: any, newData: any) => any;
+    };
+}
+export namespace Colors {
+    const yellow: string;
+    const orange: string;
+    const red: string;
+    const pink: string;
+    const purple: string;
+    const blue: string;
+    const green: string;
+    const grey: string;
+}
+export class FlumeConfig {
+    constructor(config: any);
```

Comment:
I think we should be able to type the config as well. The docs show three methods addPortType, addNodeType and addRootNodeType. They all take a config object of the corresponding shape you have described in index.d.ts. Typing these would be really helpful when configuring a flume instance.

Reply:
Hmm, not sure if I follow you here. Should this config be typed to the class `FlumeConfig`?

Reply:
Yes, the constructor takes another config (as far as I can tell that was the intent). However, it only uses a limited part of said config.

Reply:
Yes I think this is correct now. Sorry for the late reply I thought Github would notify me, but it didn't.